### PR TITLE
EZ-393 Integrating footer component

### PR DESCRIPTION
**Pull request by surbhi90**

### CODE REVIEW

File: ezcontent_pattern.theme [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+/**
+ * @file
+ *
```
Comment (vaibhavjain-in) [inline ezcontent_pattern.theme:5]: Remove this additional line.

File: ezcontent_pattern.theme [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ * Place your custom PHP code in this file.
+ */
+
+ /**
+ * Implements hook_preprocess_HOOK().
+ */
```
Comment (vaibhavjain-in) [inline ezcontent_pattern.theme:11]:
```suggestion
 /**
  * Implements hook_preprocess_HOOK().
  */
```

File: ezcontent_pattern.theme [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
```
Comment (vaibhavjain-in) [inline ezcontent_pattern.theme:12]: Remove this line.

File: templates/block/block--companyaddress-2.html.twig [hunk 4]
```diff
@@ -0,0 +1,67 @@
+{#
```
Comment (vaibhavjain-in) [inline templates/block/block--companyaddress-2.html.twig:1]: I am not sure, but why the template name has `2` here ? Does the direct name `templates/block/block--companyaddress.html.twig` not work ?

File: templates/block/block--social-media-links-block.html.twig [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    "url": item.url,
+    "markup": item.element['#markup']
+  }])
+  %}
+{% endfor %}
+{% set data = {
+  "socialmedia":{
+    "medialinks": mediaArray
+  }
+} %}
```
Comment (vaibhavjain-in) [inline templates/block/block--social-media-links-block.html.twig:72]:
```suggestion
  } 
%}
```

File: templates/block/block--social-media-links-block.html.twig [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+{% set mediaArray = [] %}
+{% for key,item in content.0['#platforms'] %}
+  {% set mediaArray = mediaArray|merge([{
+    "id": item.id,
+    "url": item.url,
+    "markup": item.element['#markup']
+  }])
+  %}
+{% endfor %}
+{% set data = {
```
Comment (vaibhavjain-in) [inline templates/block/block--social-media-links-block.html.twig:68]:
```suggestion
{% 
  set data = {
```

File: templates/block/block--system-menu-block--footer-menu.html.twig [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    "attributes": item.attributes,
+    "class": "nav-link"
+  }])
+  %}
+{% endfor %}
+{% set data = {
+  "menu":{
+    "footerTop": menuArray
+  }
+} %}
```
Comment (vaibhavjain-in) [inline templates/block/block--system-menu-block--footer-menu.html.twig:73]:
```suggestion
{% 
  set data = {
    "menu":{
      "footerTop": menuArray
    }
  } 
%}
```

**Review by vaibhavjain-in — CHANGES_REQUESTED** [review]
@surbhi90 Please see feedback

File: ezcontent_pattern.theme (hunk 1 above)
Comment (surbhi90) [reply ezcontent_pattern.theme]: done

File: ezcontent_pattern.theme (hunk 2 above)
Comment (surbhi90) [reply ezcontent_pattern.theme]: Done

File: ezcontent_pattern.theme (hunk 3 above)
Comment (surbhi90) [reply ezcontent_pattern.theme]: Done

File: templates/block/block--social-media-links-block.html.twig (hunk 5 above)
Comment (surbhi90) [reply templates/block/block--social-media-links-block.html.twig]: Done

File: templates/block/block--system-menu-block--footer-menu.html.twig (hunk 7 above)
Comment (surbhi90) [reply templates/block/block--system-menu-block--footer-menu.html.twig]: Done

File: templates/block/block--social-media-links-block.html.twig (hunk 6 above)
Comment (surbhi90) [reply templates/block/block--social-media-links-block.html.twig]: Done

File: templates/block/block--companyaddress-2.html.twig (hunk 4 above)
Comment (surbhi90) [reply templates/block/block--companyaddress-2.html.twig:1]: I have checked this but for the company address block, it is showing "block--companyaddress-2.html.twig" in suggestions and it is not taking this twig if i removed 2 from it name.

**Review by vaibhavjain-in — APPROVED** [review]
LGTM

**Pull request merged**